packages/datadog-plugin-aws-sdk/test/serverless-peer-service.spec.js

```diff
@@ -0,0 +1,319 @@
+const helpers = require('./kinesis_helpers')
+const { promisify } = require('util')
+const { setup } = require('./spec_helpers')
+const sinon = require('sinon')
+
+describe('Plugin', () => {
+  describe('Serverless', function () {
+    let tracer
+    setup()
+
+    withVersions('aws-sdk', ['aws-sdk', '@aws-sdk/smithy-client'], (version, moduleName) => {
+      let AWS
+
+      before(async () => {
+        tracer = require('../../dd-trace')
+        sinon.stub(Config.prototype, '_isInServerlessEnvironment').returns(true)
+        // force a reload of the plugins by deleting the plugin instance
+        delete tracer._pluginManager._pluginsByName['aws-sdk']
+        // load the agent and plugins again to ensure `_isInServerlessEnvironment` check
+        // in the plugin constructor is called and the necessary channels are subscribed
+        await agent.load(['aws-sdk', 'http'], [{}, { server: false }])
+      })
+
+      after(async () => {
+        Config.prototype._isInServerlessEnvironment.restore()
+        await agent.close({ ritmReset: false })
+      })
+
+      describe('DynamoDB-Serverless', () => {
+        let dynamo
+        const tableName = 'PeerServiceTestTable'
+        const dynamoClientName = moduleName === '@aws-sdk/smithy-client' ? '@aws-sdk/client-dynamodb' : 'aws-sdk'
+
+        function getCreateTableParams () {
+          return {
+            TableName: tableName,
+            KeySchema: [{ AttributeName: 'id', KeyType: 'HASH' }],
+            AttributeDefinitions: [{ AttributeName: 'id', AttributeType: 'S' }],
+            ProvisionedThroughput: { ReadCapacityUnits: 5, WriteCapacityUnits: 5 }
+          }
+        }
+
+        function toPromise (client, fn) {
+          return function (...args) {
+            if (typeof fn === 'function' && typeof fn.bind === 'function') {
+              fn = fn.bind(client)
+            }
+
+            const result = fn(...args)
+
+            if (result && typeof result.then === 'function') {
+              return result
+            }
+
+            if (result && typeof result.promise === 'function') {
+              return result.promise()
+            }
+
+            return promisify(fn)(...args)
+          }
+        }
+
+        before(async () => {
+          AWS = require(`../../../versions/${dynamoClientName}@${version}`).get()
+
+          dynamo = new AWS.DynamoDB({ endpoint: 'http://127.0.0.1:4566', region: 'us-east-1' })
+
+          // ignore error if the table already exists
+          if (typeof dynamo.createTable === 'function') {
+            const createTable = toPromise(dynamo, dynamo.createTable)
+            try { await createTable(getCreateTableParams()) } catch (_) {}
+          }
+        })
+
+        it('propagates peer.service from aws span to underlying http span', async () => {
+          const send = toPromise(dynamo, dynamo.putItem)
+
+          const tracesPromise = agent.assertSomeTraces(assertPeerServicePropagation)
```

Comment:
Can real values be asserted explicitly instead of just checking for the general structure?

Reply:
What I meant is to have the individual expectations in the test itself, with the values directly. This is for 3 reasons:

1) By checking that a value exists but not checking the value, we're not really ensuring that the values are correct.
2) Indirection in tests make them difficult to read and figure out what the expectation actually is.
3) Debugging a test is made a lot more difficult with code reuse as the same helper ends up running for every test.